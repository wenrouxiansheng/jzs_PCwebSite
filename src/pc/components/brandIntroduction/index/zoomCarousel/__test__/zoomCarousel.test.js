import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import ZoomCarousel from "../index";

describe("PC:<ZoomCarousel />", () => {
  it("render", () => {
    const wrapper = renderer.create(<ZoomCarousel />);
    let tree = wrapper.toJSON();

    //渲染测试
    expect(tree).toMatchSnapshot();
  });

  const wrapper = mount(<ZoomCarousel />);

  it("next event", () => {
    //下一张
    wrapper.find(".next").simulate("click");
    expect(wrapper.state("currentSub")).toBe(2);
  });
  it("prev event", () => {
    //上一张
    wrapper.find(".prev").simulate("click");
    //使用的同一个wrapper 状态会延续 所以是1不是0
    expect(wrapper.state("currentSub")).toBe(1);
  });
});
