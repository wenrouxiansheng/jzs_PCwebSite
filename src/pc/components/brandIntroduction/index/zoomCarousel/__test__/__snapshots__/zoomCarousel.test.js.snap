// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PC:<ZoomCarousel /> render 1`] = `
<div
  className="zoomCarousel"
  style={Object {}}
>
  <div
    className="carousel-container"
  >
    <div
      className="carousel-list"
      flag={1}
      style={
        Object {
          "transform": "translateX(0px)",
          "width": "1850px",
        }
      }
    >
      <div
        className="carousel-slide "
      >
        <img
          alt=""
          src="test-file-stub"
        />
      </div>
      <div
        className="carousel-slide active"
      >
        <img
          alt=""
          src="test-file-stub"
        />
      </div>
      <div
        className="carousel-slide "
      >
        <img
          alt=""
          src="test-file-stub"
        />
      </div>
      <div
        className="carousel-slide "
      >
        <img
          alt=""
          src="test-file-stub"
        />
      </div>
      <div
        className="carousel-slide "
      >
        <img
          alt=""
          src="test-file-stub"
        />
      </div>
    </div>
  </div>
  <div
    className="button prev"
    onClick={[Function]}
  >
    <span
      aria-label="left"
      className="anticon anticon-left"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="left"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
        />
      </svg>
    </span>
  </div>
  <div
    className="button next"
    onClick={[Function]}
  >
    <span
      aria-label="right"
      className="anticon anticon-right"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="right"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
        />
      </svg>
    </span>
  </div>
</div>
`;
